### PR TITLE
fix(osc):add version whitelist  to enable lock table feature

### PR DESCRIPTION
**Pull request by LioRoger**
…range is 4.2.[5 - 9]

#### What type of PR is this?

type-bug
#### What this PR does / why we need it:
1. lock table feature for ObMySQL version is [4.2.5 ~ 4.3.0) for now.
2. previous logic will enable lock table for Ob version greater than 4.2.5 which have no rollback control for 4.3.x
3. introduce white list format in json array, default value is  [4.2.[5 - 9].*].  matched with java regex lib.
4. value is stored in config_system_configuration with key "odc.osc.support-lock-table-ob-version-json" , use sql like :"insert into config_system_configuration(`key`, `application`, `profile`, `label`, `description`, value) values('odc.osc.support-lock-table-ob-version-json', 'odc', 'default', 'master', 'control odc enable lock table version, default is ["4\\\\.2\\\\.5.*"]', '["4\\\\.2\\\\.5.*"]')"    to init it. 
5. if config_system_configuration has no rule, default value in memory is "4\.2\.[5-9].*."
6. ODC upgrade will not affect previous stored value.
 
#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/onlineschemachange/rename/LockTableSupportDecider.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * @author longpeng.zlp
+ * @date 2024/12/2 16:30
+ */
+@Slf4j
+public class LockTableSupportDecider {
+    public static final LockTableSupportDecider DEFAULT_LOCK_TABLE_DECIDER =
+            new LockTableSupportDecider(Arrays.asList("4\\.2\\.[5-9].*"));
```
Comment (MarkPotato777) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/onlineschemachange/rename/LockTableSupportDecider.java:36]: We have `VersionUtils` to compare versions.

Comment (LioRoger) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/onlineschemachange/rename/LockTableSupportDecider.java:36]: this is regex match, not accurate version compare

**Review by MarkPotato777 — APPROVED** [review]
LGTM

**Review by LuckyPickleZZ — COMMENTED** [review]
lgtm

**Review by LuckyPickleZZ — APPROVED** [review]

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**